src/app/main.rs

```diff
@@ -474,14 +476,32 @@ where
             //    is retryable.
             let dst_route_layer = svc::builder()
                 .layer(buffer::layer(max_in_flight))
-                .layer(pending::layer())
+                .layer(pending::layer::<_, dst::Route, _>())
                 .layer(classify::layer())
                 .layer(metrics::layer::<_, classify::Response>(route_http_metrics))
                 .layer(proxy::http::timeout::layer())
                 .layer(retry::layer(retry_http_metrics.clone()))
                 .layer(metrics::layer::<_, classify::Response>(retry_http_metrics))
                 .layer(insert_target::layer());
 
+            let balancer = svc::builder()
+                .layer(
+                    balance::layer(
+                        EWMA_DEFAULT_RTT,
+                        EWMA_DECAY,
+                        resolve::layer(Resolve::new(resolver)),
+                    )
+                    .with_fallback(max_in_flight, |req: &http::Request<_>| {
```

Comment:
having to pass the in-flight limit here is a little unfortunate, but it's necessary as the buffer is only applied to the stack under the router (and not to the stack under the lb)

Reply:
The buffer is only necessary so that we have a cloneable sender, not to actually hold requests in a queue... Can we Use a Buffer with a capacity of 1 to force backpressure? when the route is in use?

Reply:
I double-checked the `tower-buffer` behaviour and that should work. Will do.

Reply:
As we discussed in person, we decided that pushing backpressure into the executor is not how we want to handle this, so instead we should be willing to buffer here.

Reply:
yup, 00d0c47433b585f32f4109d18814885d4a1f60ac reverted the change that added a capacity 1 buffer.